native_lib/DDGCmdline.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+#include <sodium.h>
+#include "DDGSyncCrypto.h"
+
+int main(int argc, char **argv) {
+    if (argc != 3) {
+        printf("Need 2 args, got %d\n", argc);
+        return 1;
+    }
+
+    unsigned char primaryKey[DDGSYNCCRYPTO_PRIMARY_KEY_SIZE];
+    unsigned char secretKey[DDGSYNCCRYPTO_SECRET_KEY_SIZE];
+    unsigned char protectedSecretKey[DDGSYNCCRYPTO_PROTECTED_SECRET_KEY_SIZE];
+    unsigned char passwordHash[DDGSYNCCRYPTO_HASH_SIZE];
+    
+    DDGSyncCryptoResult res;
+
+    res = ddgSyncGenerateAccountKeys(primaryKey, secretKey, protectedSecretKey, passwordHash, argv[1], argv[2]);
```

Comment:
I don't think you can use `ddgSyncGenerateAccountKeys` as is. The script *MUST* use a fixed `secretKey` in order for the pre-encrypted bookmarks and credentials to be valid, and `ddgSyncGenerateAccountKeys` generates a new random `secretKey`.

Reply:
I think that a cli wrapper around functions like `crypto_secretbox_easy` and `crypto_kdf_derive_from_key` would be better than wrapping around `DDGSyncCrypto.c`. I'd do that rather than trying to refactor `ddgSyncGenerateAccountKeys`.

Reply:
I think it's a good idea, I can circle back to this.  For the moment a purpose-built tool is the "low hanging fruit".